src/main/java/sparksoniq/spark/SparkSessionManager.java

```diff
@@ -32,55 +13,51 @@
 import sparksoniq.jsoniq.tuple.FlworTuple;
 import sparksoniq.semantics.DynamicContext;
 
-public class SparkContextManager {
+public class SparkSessionManager {
+
+    private static SparkSessionManager _instance;
+    private SparkConf configuration;
+    private SparkSession session;
+    private JavaSparkContext javaSparkContext;
 
     public static int COLLECT_ITEM_LIMIT = 0;
     private static Level LOG_LEVEL = Level.FATAL;
     private static final String APP_NAME = "jsoniq-on-spark";
 
+
     public static boolean LIMIT_COLLECT() {
         return COLLECT_ITEM_LIMIT > 0;
     }
 
-    public static SparkContextManager getInstance() {
-
+    public static SparkSessionManager getInstance() {
         if (_instance == null)
-            _instance = new SparkContextManager();
+            _instance = new SparkSessionManager();
         return _instance;
     }
 
-    private static SparkContextManager _instance;
-
-    private SparkContextManager() {
+    private SparkSessionManager() {
     }
 
-    public void initializeConfigurationAndContext() {
-        configuration = new SparkConf().setAppName(APP_NAME);
-        initialize();
-    }
-
-    public void initializeConfigurationAndContext(SparkConf conf, boolean setAppName) {
-        if (setAppName)
-            conf.setAppName(APP_NAME);
-        configuration = conf;
-        initialize();
-    }
-
-    public JavaSparkContext getContext() {
-        if (context == null) {
-            if (this.configuration == null)
-                initializeConfigurationAndContext();
+    public SparkSession getSession() {
+        if (session == null) {
+            if (this.configuration == null) {
+                setDefaultConfiguration();
+            }
             initialize();
         }
-        return context;
+        return session;
+    }
+
+    private void setDefaultConfiguration () {
+        configuration = new SparkConf().setAppName(APP_NAME);
     }
 
     private void initialize() {
         initializeKryoSerialization();
         Logger.getLogger("org").setLevel(LOG_LEVEL);
         Logger.getLogger("akka").setLevel(LOG_LEVEL);
-        if (context == null)
-            context = new JavaSparkContext(this.configuration);
+        if (session == null)
+            session= SparkSession.builder().config(this.configuration).getOrCreate();
```

Comment:
Can we throw an error if the session already exists? I am afraid that otherwise we can run into inconsistent states (if the public initialization functions are called with several configurations).